pgx-tests/src/tests/pg_guard_tests.rs

```diff
@@ -0,0 +1,46 @@
+/*
+Portions Copyright 2019-2021 ZomboDB, LLC.
+Portions Copyright 2021-2022 Technology Concepts & Design, Inc. <support@tcdi.com>
+
+All rights reserved.
+
+Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+*/
+
+use pgx::prelude::*;
+
+#[pg_extern]
+fn extern_func() -> bool {
+    extern_func_impl::<u8>()
+}
+
+// This ensures that parameterized function compiles when it has `pg_guard` attached to it
+#[pg_guard]
+// Uncommenting the line below will make it fail to compile
+// #[no_mangle]
```

Comment:
We should have some kind of compiletest-like suite, so we can test for compile failures.